From 4fb766c59c717197cb44af35b634392cd9826779 Mon Sep 17 00:00:00 2001
Message-ID: <4fb766c59c717197cb44af35b634392cd9826779.1732837202.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732837202.git.sharlatanus@gmail.com>
References: <cover.1732837202.git.sharlatanus@gmail.com>
From: "Artyom V. Poptsov" <poptsov.artyom@gmail.com>
Date: Sun, 24 Nov 2024 13:27:26 +0300
Subject: [PATCH 05/15] gnu: Add go-github-com-sahilm-fuzzy.

* gnu/packages/golang-xyz.scm (go-github-com-sahilm-fuzzy): New variable.

Change-Id: Iebf4f8c3848e1e08b2860ac69a5a8d816ac260df
Reviewed-by: Sharlatan Hellseher <sharlatanus@gmail.com>
---
 gnu/packages/golang-xyz.scm | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index fefc5c6a97..2ff995bb14 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -6666,6 +6666,32 @@ (define-public go-github-com-ryanuber-columnize
      "This package implements column-formatted output for Golang.")
     (license license:expat)))
 
+(define-public go-github-com-sahilm-fuzzy
+  (package
+    (name "go-github-com-sahilm-fuzzy")
+    (version "0.1.1")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/sahilm/fuzzy")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "15j95gm7hcmg09x1b39vc4il8bryv4v0yljvvyq5vyc6iq66qrbz"))))
+    (build-system go-build-system)
+    (native-inputs
+     (list go-github-com-kylelemons-godebug
+           go-github-com-kylelemons-godebug-pretty))
+    (arguments
+     (list #:import-path "github.com/sahilm/fuzzy"))
+    (home-page "https://github.com/sahilm/fuzzy")
+    (synopsis "Fuzzy string matching for Golang")
+    (description
+     "@code{fuzzy} provides fuzzy string matching optimized for filenames and code
+symbols in the style of Sublime Text, VSCode, @code{IntelliJ} IDEA et al.")
+    (license license:expat)))
+
 (define-public go-github-com-schollz-progressbar-v3
   (package
     (name "go-github-com-schollz-progressbar-v3")
-- 
2.46.0

